mem_context.go

```diff
@@ -51,6 +51,14 @@ func (dt *depthTracker) Ascend() {
 	dt.curDepth--
 }
 
+type limitChecker struct {
```

Comment:
why is this its own struct?

Reply:
It didn't need to be but I was foreseeing us needing to have other limits/thresholds to check. We can revisit the name but I thought we could use it to hold the array len threshold as well so we can tweak it from baas (like you mentioned).